app/src/main/java/com/android/shelfLife/ui/overview/AddFoodItem.kt

```diff
@@ -0,0 +1,217 @@
+                            onClick = {
+                              location = selectionOption
+                              locationExpanded = false
+                            })
+                      }
+                    }
+              }
+
+          // For dates a future improvement could be having a calendar interfare rather than manual
+          // input
+          OutlinedTextField(
+              value = expireDate,
+              onValueChange = { expireDate = it },
+              label = { Text("Expire Date") },
+              placeholder = { Text("dd/mm/yyyy") },
+              modifier = Modifier.fillMaxWidth().padding(bottom = 16.dp),
+          )
+
+          OutlinedTextField(
+              value = openDate,
+              onValueChange = { openDate = it },
+              label = { Text("Open Date") },
+              placeholder = { Text("dd/mm/yyyy") },
+              modifier = Modifier.fillMaxWidth().padding(bottom = 16.dp),
+          )
+
+          OutlinedTextField(
+              value = buyDate,
+              onValueChange = { buyDate = it },
+              label = { Text("Buy Date") },
+              placeholder = { Text("dd/mm/yyyy") },
+              modifier = Modifier.fillMaxWidth().padding(bottom = 32.dp),
+          )
+
+          Button(
+              onClick = {
+                val foodFacts =
+                    FoodFacts(
+                        name = foodName,
+                        barcode = "",
+                        quantity = Quantity(amount.toDouble(), unit),
+                        category = category,
+                        // nutritionFacts = NutritionFacts()
+                    )
+                val newFoodItem =
+                    FoodItem(
+                        uid = foodItemViewModel.getUID(),
+                        foodFacts = foodFacts,
+                        location = location,
+                        expiryDate = formatDateToTimestamp(expireDate),
+                        openDate = formatDateToTimestamp(openDate),
+                        buyDate = formatDateToTimestamp(buyDate),
+                        // Have to add logic to determine status depending on the dates given here/
+                        // should be calculated
+                        status = FoodStatus.CLOSED)
+                houseHoldViewModel.addFoodItem(newFoodItem)
+                navigationActions.goBack()
+              },
+              modifier = Modifier.fillMaxWidth().height(50.dp)) {
+                Text(text = "Submit", fontSize = 18.sp)
+              }
+        }
+  }
+}
+
+fun formatTimestampToDate(timestamp: Timestamp): String {
+  val sdf = SimpleDateFormat("dd/MM/yyyy", Locale.getDefault())
+  return sdf.format(timestamp.toDate())
+}
+
+fun formatDateToTimestamp(dateString: String): Timestamp {
+  val sdf = SimpleDateFormat("dd/MM/yyyy", Locale.getDefault())
+  val date = sdf.parse(dateString)
+  return Timestamp(date)
+}
```

Comment:
Are those two functions only used in this file ? If yes they should probably be marked as `private` to prevent anyone else from accessing  them.

If they are not and are intended to be used by other `composable`, `viewModel` or `repositories`, I would advise to move them into another file where such "util" function could be written. It will prevent anyone from defining multiple times the same function.